nabavka/src/main/java/raf/si/racunovodstvo/nabavka/controllers/KalkulacijaController.java

```diff
@@ -0,0 +1,76 @@
+import raf.si.racunovodstvo.nabavka.services.impl.KalkulacijaService;
+import raf.si.racunovodstvo.nabavka.utils.ApiUtil;
+import raf.si.racunovodstvo.nabavka.utils.SearchUtil;
+
+import javax.persistence.EntityNotFoundException;
+import javax.validation.Valid;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import java.util.Optional;
+
+@CrossOrigin
+@RestController
+@SecurityRequirement(name = "bearerAuth")
+@RequestMapping("/api/kalkulacije")
+public class KalkulacijaController {
+
+    private final KalkulacijaService kalkulacijaService;
+
+    private final SearchUtil<Kalkulacija> searchUtil;
+
+    public KalkulacijaController(KalkulacijaService kalkulacijaService) {
+        this.kalkulacijaService = kalkulacijaService;
+        this.searchUtil = new SearchUtil<>();
+    }
+
+    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> searchKalkulacija(
+        @RequestParam(name = "search", required = false, defaultValue = "") String search,
+        @RequestParam(defaultValue = ApiUtil.DEFAULT_PAGE) @Min(ApiUtil.MIN_PAGE) Integer page,
+        @RequestParam(defaultValue = ApiUtil.DEFAULT_SIZE) @Min(ApiUtil.MIN_SIZE) @Max(ApiUtil.MAX_SIZE) Integer size,
+        @RequestParam(defaultValue = "brojKalkulacije") String[] sort
+    ) {
+        Pageable pageSort = ApiUtil.resolveSortingAndPagination(page, size, sort);
+        Specification<Kalkulacija> spec = this.searchUtil.getSpec(search);
+        return ResponseEntity.ok(this.kalkulacijaService.findAll(spec, pageSort));
+    }
+
+    @GetMapping(value = "/all", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getKalkulacija() {
+        return ResponseEntity.ok(kalkulacijaService.findAll());
+    }
+
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> createKalkulacija(@Valid @RequestBody KalkulacijaRequest kalkulacija) {
+        return ResponseEntity.ok(kalkulacijaService.save(kalkulacija));
+    }
+
+    @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> updateKalkulacija(@Valid @RequestBody KalkulacijaRequest kalkulacija) {
+        Optional<Kalkulacija> optionalKalkulacija = kalkulacijaService.findById(kalkulacija.getId());
+        if(optionalKalkulacija.isPresent()) {
+            return ResponseEntity.ok(kalkulacijaService.save(kalkulacija));
+        }
+        throw new EntityNotFoundException();
+    }
+
+    @DeleteMapping(value = "/{id}")
+    public ResponseEntity<?> deleteKalkulacija(@PathVariable("id") Long id){
+        Optional<Kalkulacija> optionalKalkulacija = kalkulacijaService.findById(id);
+        if (optionalKalkulacija.isPresent()){
+            kalkulacijaService.deleteById(id);
+            return ResponseEntity.noContent().build();
+        }
+        throw new EntityNotFoundException();
```

Comment:
Mozda bi ova logika mogla da se prebaci u servis, umesto da se vrsi provera u kontroleru.

Reply:
U celom backendu je napisano na ovaj nacin, mozemo eventualno sve da refaktorisemo ali to bi bio drugi task